pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -390,6 +394,20 @@ func checkRetry(ctxPtr *zedmanagerContext) {
 	}
 }
 
+func checkDelayedStartApps(ctx *zedmanagerContext) {
+	configs := ctx.subAppInstanceConfig.GetAll()
+	for _, c := range configs {
+		config := c.(types.AppInstanceConfig)
+		status := lookupAppInstanceStatus(ctx, config.Key())
+		if status != nil && status.State == types.START_DELAYED && config.StartTime.Before(time.Now()) && !status.DelayHandled {
+			log.Warnf("@ohm: Time to doUpdate!")
+			status.DelayHandled = true
```

Comment:
It seems that in current logic if we will expand delay time of the app that passed delay (to be more than time passed from the boot) and will restart app it will not started eventually because of DelayHandled. So it seems reasonable to clear DelayHandled if we receive modified configuration with config.StartTime more than we have.
Also please add comments about our expectations.

Reply:
Do you think we should apply the delay in the case of the App restart?

Reply:
Well, good question, but we do not explicitly block this possibility now. I think we can set DelayHandled to false in all cases we receive config with StartTime after time.Now() just to ensure that eventually we will handle it inside checkDelayedStartApps.

Reply:
Done